modules/storage/src/service/compression/mod.rs

```diff
@@ -22,6 +23,17 @@ impl Display for Compression {
     }
 }
 
+impl FromStr for Compression {
```

Comment:
I think we already use `strum`, which should be able to do the same.

Reply:
Updated. Got a little carried away handling the potential `ParseError` though.